apps/server/src/modules/tool/external-tool/service/external-tool-logo.service.ts

```diff
@@ -77,14 +78,22 @@ export class ExternalToolLogoService {
 			this.logger.info(new ExternalToolLogoFetchedLoggable(logoUrl));
 
 			const buffer: Buffer = Buffer.from(response.data);
+			this.detectContentType(buffer);
+
 			const logoBase64: string = buffer.toString('base64');
 
 			return logoBase64;
 		} catch (error) {
-			if (error instanceof HttpException) {
-				throw new ExternalToolLogoFetchFailedLoggableException(logoUrl, error.getStatus());
-			} else {
-				throw new ExternalToolLogoFetchFailedLoggableException(logoUrl);
+			switch (error) {
```

Comment:
Move this in a private function and use if- statements instead of switch and throw:

if (error instanceof HttpException) {
      throw new ExternalToolLogoFetchFailedLoggableException(logoUrl, error.getStatus());
    } else if (error instanceof ExternalToolLogoWrongFileTypeLoggableException) {
      throw error;
    } else {
      throw new ExternalToolLogoFetchFailedLoggableException(logoUrl);
    }